From 80a662c4cdfbaa068a68d04613334cf0840f35bc Mon Sep 17 00:00:00 2001
From: Harman Kalra <hkalra@marvell.com>
Date: Tue, 6 Apr 2021 14:47:48 +0530
Subject: [PATCH 1398/1921] octeontx2-af: fix multiple entries for vlan lt defs

Due to multiple entries of ovlan/ivlan lt defs in
struct npc_lt_def_cfg, KPU offset calculation while NPC
profile loading results in wrong offsets and hence profile
loading via fwdb fails.

Fixes: 860f3d5f81e4 ("octeontx2-af: cn10k: Bandwidth profiles config support")

Change-Id: I41840dd00494d250dde5a4da3c7f98f6c9de4645
Signed-off-by: Harman Kalra <hkalra@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49309
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <SunilKovvuri.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc.h b/drivers/net/ethernet/marvell/octeontx2/af/npc.h
index c23cd370116a..231cdee8f165 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc.h
@@ -528,8 +528,6 @@ struct npc_lt_def_et {
 
 struct npc_lt_def_cfg {
 	struct npc_lt_def	rx_ol2;
-	struct npc_lt_def	ovlan;
-	struct npc_lt_def	ivlan;
 	struct npc_lt_def	rx_oip4;
 	struct npc_lt_def	rx_iip4;
 	struct npc_lt_def	rx_oip6;
@@ -547,8 +545,8 @@ struct npc_lt_def_cfg {
 	struct npc_lt_def	pck_iip4;
 	struct npc_lt_def_apad	rx_apad0;
 	struct npc_lt_def_apad	rx_apad1;
-	struct npc_lt_def_color	rx_ovlan;
-	struct npc_lt_def_color	rx_ivlan;
+	struct npc_lt_def_color	ovlan;
+	struct npc_lt_def_color	ivlan;
 	struct npc_lt_def_color	rx_gen0_color;
 	struct npc_lt_def_color	rx_gen1_color;
 	struct npc_lt_def_et	rx_et[2];
-- 
2.31.1

